### PR TITLE
Moe sync

### PR DESCRIPTION
**Pull request by ronshapiro**

### CODE REVIEW

File: idea_plugin/src/com/google/googlejavaformat/intellij/FormatterUtil.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    try {
+      ImmutableMap.Builder<TextRange, String> replacements = ImmutableMap.builder();
+      formatter
+          .getFormatReplacements(text, toRanges(ranges))
+          .forEach(
+              replacement ->
+                  replacements.put(
+                      toTextRange(replacement.getReplaceRange()),
+                      replacement.getReplacementString()));
+      return replacements.build();
```
Comment (jbduncan) [inline idea_plugin/src/com/google/googlejavaformat/intellij/FormatterUtil.java]: @plumpy I believe that the code in this try statement could be made more idiomatic with something like:

```java
return formatter
    .getFormatReplacements(text, toRanges(ranges))
    .stream()
    .collect(
        toImmutableMap(
            replacement -> toTextRange(replacement.getReplaceRange()),
            replacement -> replacement.getReplacementString()));
```

Comment (plumpy) [reply idea_plugin/src/com/google/googlejavaformat/intellij/FormatterUtil.java]: This code is designed to run in IDEA 2017.1+, but 2017.1 has Guava 19, which does not have any Java 8 features like toImmutableMap.

Comment (plumpy) [reply idea_plugin/src/com/google/googlejavaformat/intellij/FormatterUtil.java]: (and I'd happily bump to 2017.2 as a minimum version, but the current version of Android Studio is based on 2017.1, so I'd like to keep supporting that)

Comment (jbduncan) [reply idea_plugin/src/com/google/googlejavaformat/intellij/FormatterUtil.java]: Ah I see, that's a bit of a shame. Thanks for the clarification. :)

**Pull request merged**